File: windows/WallpaperInfoApp/WallpaperInfoApp/WallpaperServiceConnection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WallpaperInfoApp {

	class WallpaperServiceConnection {

		WallpaperServiceConnection(WallpaperInfoUI wallpaperInfoUI ) {
			_wallpaperInfoUI = wallpaperInfoUI;
		}
	
		public static void startService() {
			WallpaperServiceHandler.startService();
		}
    
		public static void stopService() {
			WallpaperServiceHandler.stopService();
		}

		public static void broadcastToService(int action, int extra = 0) {
			WallpaperServiceInfo.getInstance().getWallpaperService().broadcastReceiver(action, extra);
		}
	
		public void sendMessageToService(int command, int intOption = 0, object objectOption = null) {
			if (_wallpaperServiceHandler != null) {
				_wallpaperServiceHandler.handleMessage(command, intOption, objectOption);
			}
		}
	
		public static WallpaperServiceConnection bindService(WallpaperInfoUI wallpaperInfoUI) {
			WallpaperServiceConnection wallpaperServiceConnection = new WallpaperServiceConnection(wallpaperInfoUI);
			wallpaperServiceConnection._wallpaperServiceHandler = WallpaperServiceHandler.incomingHandler(incomingMessenger: wallpaperServiceConnection);
			return wallpaperServiceConnection;
		}
	
		public void unbindService() {
			_wallpaperServiceHandler.unbind();
			_wallpaperServiceHandler = null;
		}

		public WallpaperInfoUI _wallpaperInfoUI;
		private WallpaperServiceHandler _wallpaperServiceHandler = null;
	}
}

File: apps/mobile-ios/tromso/tromso/WallpaperServiceConnection.swift
//
//  WindowServiceConnection.swift
//  WallpaperInfoApp
//
//  Created by Brad Park on 4/15/20.
//  Copyright © 2020 Brad Park. All rights reserved.
//

import Foundation

class WallpaperServiceConnection {

	init(_ contentView: ContentView) {
		_delegate = contentView
	}
	
    public static func startService() {
        WallpaperServiceHandler.startService()
    }
    
    public static func stopService() {
        WallpaperServiceHandler.stopService()
    }
    
	public static func broadcastToService(_ action:Int, extras:Any? = nil) {
        WallpaperServiceInfo.getInstance().getWallpaperService().broadcastReceiver(action, extras)
    }
	
    func sendMessageToService(command:Int, intOption:Int = 0, objectOption:Any? = nil) {
		if _wallpaperServiceHandler != nil {
			_wallpaperServiceHandler?.handleMessage(command, intOption, objectOption)
		}
    }
	
	class func bindService(contentView: ContentView) -> WallpaperServiceConnection {
		let wallpaperServiceConnection = WallpaperServiceConnection(contentView)
		wallpaperServiceConnection._wallpaperServiceHandler = WallpaperServiceHandler.incomingHandler(incomingMessenger: wallpaperServiceConnection)
		return wallpaperServiceConnection
	}
	
	func unbindService() {
		_wallpaperServiceHandler?.unbind()
		_wallpaperServiceHandler = nil
	}

	var _delegate: ContentView
	private var _wallpaperServiceHandler: WallpaperServiceHandler? = nil
}
